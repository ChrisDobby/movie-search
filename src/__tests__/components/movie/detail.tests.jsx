import React from 'react';
import renderer from 'react-test-renderer';
import Detail from '../../../components/movie/detail';

describe('movieCard', () => {
    const movie = {
        id: 1,
        title: 'A Movie',
        tagLine: 'some tagline',
        overview: 'something',
        posterImage: '1.png',
        genres: ['Action'],
    };

    it('should render the movie detail correctly', () => {
        const detail = renderer.create(<Detail data={movie} actions={{}} />);

        expect(detail.toJSON()).toMatchSnapshot();
    });
});
