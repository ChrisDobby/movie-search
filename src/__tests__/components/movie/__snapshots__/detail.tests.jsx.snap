// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movieCard should render the movie detail correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-8 col-md-7 col-sm-6 col-xs-12"
  >
    <h1>
      A Movie
    </h1>
    <h3>
      some tagline
    </h3>
    <div
      style={
        Object {
          "backgroundColor": "#e4e4e4",
          "borderRadius": "16px",
          "color": "rgba(0,0,0,0.6)",
          "display": "inline-block",
          "fontSize": "13px",
          "fontWeight": "500",
          "height": "32px",
          "lineHeight": "32px",
          "marginBottom": "5px",
          "marginRight": "5px",
          "padding": "0 12px",
        }
      }
    >
      Action
    </div>
    <p>
      something
    </p>
    <div>
      <h4>
        Rate this movie
      </h4>
      <button
        className="btn btn-link"
        onClick={[Function]}
        style={
          Object {
            "whiteSpace": "normal",
          }
        }
      >
        You are not logged in to MovieDb.  Click here to login.
      </button>
    </div>
  </div>
  <div
    className="col-lg-4 col-md-5 col-sm-6 col-xs-12"
  >
    <img
      alt="A Movie"
      src="1.png"
      style={
        Object {
          "borderRadius": "8px",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
